### PR TITLE
Migrate PR internal check healthcare shared components

### PR DESCRIPTION
**Pull request by v-imazumder**
## Description
Triggers: PR validation any branch

## Related issues
Addresses [issue #].

## Testing
Describe how this change was tested.

## Semver Change ([docs](https://github.com/microsoft/healthcare-shared-components/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
```
Comment (LTA-Thinking) [inline global.json:1]: The global.json changes shouldn't be needed in this review since they were part of the CI pipeline review. Merging from the master branch should remove them from the PR.

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+jobs:
+
+- job: Phase_1
+  displayName: Phase 1
+  cancelTimeoutInMinutes: 1
+  pool:
+    vmImage: $(vmImage)
+
+  steps:
+  - checkout: self
```
Comment (wsugarman) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:26]: Let's not use NuGet unless we really need it. Instead can we rely only on `dotnet`? `dotnet build` should take care of restoring as well

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+parameters:
+- name: BuildConfiguration
+  default: release
+
+variables:
+- name:  vmImage
+  value: windows-latest
+
+resources:
```
Comment (wsugarman) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:11]: This isn't necessary, is it?

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,149 @@
+name:  pr$(System.PullRequest.PullRequestNumber)-$(Date:yyyyMMdd)$(Rev:-r)
+
+parameters:
+- name: BuildConfiguration
+  default: release
+
+variables:
```
Comment (wsugarman) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:7]: Do we need to run on Windows? Is it because of component governance?

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    enabled: False
+
+  - task: AutoApplicability@1
+    displayName: Run AutoApplicability
+    continueOnError: True
+    inputs:
+      ExternalRelease: true
+      IsService: true
+
+  - task: BinSkim@3
```
Comment (wsugarman) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:71]: This is for native code isn't it?

File: nuget.config [hunk 6]
```diff
@@ -8,6 +8,7 @@
     <clear />
     <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
     <add key="Microsoft Health OSS" value="https://microsofthealthoss.pkgs.visualstudio.com/FhirServer/_packaging/Public/nuget/v3/index.json" />
+    <add key="Internalfeed" value="https://pkgs.dev.azure.com/microsofthealth/_packaging/Internalfeed/nuget/v3/index.json" />
```
Comment (wsugarman) [inline nuget.config:11]: Are we reading from the internal feed?

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 7]
```diff
@@ -0,0 +1,149 @@
+name:  pr$(System.PullRequest.PullRequestNumber)-$(Date:yyyyMMdd)$(Rev:-r)
```
Comment (wsugarman) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:1]: I think we can reduce the number of the checks we run on our code as they probably haven't been re-evaluated in awhile. For example, in DICOM we use the following: https://github.com/microsoft/dicom-server/blob/main/build/analyze.yml

And since we're running on Windows, you don't need to do the copy work we do to take the binaries we built on Linux machines.

**Review by wsugarman — CHANGES_REQUESTED** [review]

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml (hunk 7 above)
Comment (v-imazumder) [reply build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:1]: @wsugarman Great suggestion. I tried to look into reducing the number of checks.... just excluded security analysis check BinSkim as per you suggestion. Could not change the  pool name as checks are configured to run on windows machine.

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    continueOnError: True
+    inputs:
+      toolVersionV2: LatestPreRelease
+      debugMode: false
+      folderSuppression: false
+
+  - task: PoliCheck@1
+    displayName: Run PoliCheck
+    continueOnError: True
+    enabled: False
```
Comment (LTA-Thinking) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:70]: Since this is disabled can it be removed?

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    continueOnError: True
+    inputs:
+      BinSkim: true
+      CredScan: true
+
+  - task: PublishSecurityAnalysisLogs@2
+    displayName: Publish Security Analysis Logs
+    condition: succeededOrFailed()
+    continueOnError: True
+    enabled: False
```
Comment (LTA-Thinking) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:88]: Same as above, since it is disabled can it be removed?

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+  - task: PostAnalysis@1
+    displayName: Post Analysis
+    condition: succeededOrFailed()
+    inputs:
+      BinSkim: true
+      CredScan: true
+
+- job: Phase_2
```
Comment (LTA-Thinking) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:97]: This whole job can be removed, nothing is enabled.

File: global.json [hunk 11]
```diff
@@ -1,5 +1,6 @@
...
     "sdk": {
+
```
Comment (wsugarman) [inline global.json:3]: Please remove newline

File: nuget.config [hunk 12]
```diff
@@ -15,4 +15,4 @@
   <disabledPackageSources>
     <add key="Microsoft Visual Studio Offline Packages" value="true" />
   </disabledPackageSources>
-</configuration>
\ No newline at end of file
+</configuration>
```
Comment (wsugarman) [inline nuget.config:18]: Please add trailing newline again

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+  - task: AutoApplicability@1
+    displayName: Run AutoApplicability
+    continueOnError: True
+    inputs:
+      ExternalRelease: true
+      IsService: true
+
+  - task: CodeMetrics@1
+    displayName: 'Run CodeMetrics '
```
Comment (wsugarman) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:50]: nit: extra space (or remove quotes)

**Review by wsugarman — APPROVED** [review]

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+      projects: '**/*.sln'
+      selectOrConfig: config
+      nugetConfigPath: nuget.config
+
+  - task: DotNetCoreCLI@1
+    name: ''
+    displayName: dotnet build
+    inputs:
+      projects: '**/*.sln'
+      arguments: --configuration ${{ parameters.BuildConfiguration }} --version-suffix $(build.buildnumber)
```
Comment (wsugarman) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:40]: Wait -- we're not running any tests in this pipeline! lol

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+  - task: DotNetCoreCLI@2
+    displayName: dotnet restore
+    inputs:
+      command: restore
+      projects: '**/*.sln'
+      selectOrConfig: config
+      nugetConfigPath: nuget.config
+
+  - task: DotNetCoreCLI@1
+    name: ''
```
Comment (wsugarman) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:36]: Remove blank name

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+- name: BuildConfiguration
+  default: release
+
+variables:
+- name:  vmImage
+  value: windows-latest 
+
+jobs:
+
+- job: Phase_1
```
Comment (wsugarman) [inline build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:13]: Perhaps a better name lol

**Review by wsugarman — CHANGES_REQUESTED** [review]

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml (hunk 14 above)
Comment (v-imazumder) [reply build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:40]: @wsugarman I did get your comments. Could you please clarify? Thanks!

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml (hunk 13 above)
Comment (v-imazumder) [reply build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:50]: Removed. Thanks

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml (hunk 16 above)
Comment (v-imazumder) [reply build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:13]: Changed to Internal_checks. Thanks :)

File: build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml (hunk 14 above)
Comment (wsugarman) [reply build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:40]: We should be running unit tests in our pipeline, right? We should be using dotnet test to ensure all of our unit tests are passing

Comment (LTA-Thinking) [reply build/.vsts-PR-internal-check-healthcare-shared-components-azureBuild-pipeline.yml:40]: The unit tests are run as part of a separate pipeline (Github action technically): https://github.com/microsoft/healthcare-shared-components/runs/4144107071?check_suite_focus=true

**Review by LTA-Thinking — APPROVED** [review]

File: nuget.config [hunk 17]
```diff
@@ -15,4 +15,7 @@
   <disabledPackageSources>
     <add key="Microsoft Visual Studio Offline Packages" value="true" />
   </disabledPackageSources>
-</configuration>
\ No newline at end of file
+
```
Comment (wsugarman) [inline nuget.config:18]: lol now there are too many newlines. Can you revert the changes in this file?

**Review by wsugarman — APPROVED** [review]
